packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js

```diff
@@ -266,6 +300,66 @@ class LightCurveViewer extends Component {
     this.d3svg.call(addInterfaceLayer)
     this.d3interfaceLayer = this.d3svg.select('.interface-layer')
     this.d3interfaceLayer.call(this.zoom)
+    this.updateInteractionMode(props.interactionMode)
+  }
+  
+  /*
+  Updates interaction logic, switching between navigation and annotation.
+   */
+  updateInteractionMode(interactionMode = '') {
+    if (!this.zoom || !this.d3interfaceLayer) return
+    
+    if (interactionMode === 'annotate') {
+      // HACK: Prevent zoom by "running in place"
+      // this.zoom.on('zoom', null) doesn't work, because transforms are just
+      // "deferred" until Move Mode is reinstated.
+      /*this.savedTransform = this.getCurrentTransform()
+      this.zoom.on('zoom', () => {
+        if (d3.event.transform.x !== this.savedTransform.x
+            || d3.event.transform.y !== this.savedTransform.y
+            || d3.event.transform.k !== this.savedTransform.k) {
+          this.zoom.transform(this.d3interfaceLayer, this.savedTransform)
+        }
+      })
+      this.d3interfaceLayer.style('cursor', 'crosshair')
+      this.d3interfaceLayer.on('click', this.doInsertAnnotation.bind(this))*/
+      
+      this.d3svg.on('click', this.doInsertAnnotation.bind(this))
+      this.d3interfaceLayer.style('display', 'none')
+      
+    } else if (interactionMode === 'move') {
+      //this.zoom.on('zoom', this.doZoom.bind(this))
+      
+      this.d3svg.on('click', null)
+      this.d3interfaceLayer.style('display', null)
+      
+    } else {  // Users should never reach this point
+      console.error('LightCurveViewer: illogical move/annotate state detected.')
+    }
+  }
+  
+  doZoom () {
+    const t = this.getCurrentTransform()
```

Comment:
Perhaps name this `currentTransform` for ease of reading for future/other devs that may work on this?

Reply:
👌 No problem.

This is just a case of me usually naming my methods as verbs instead of nouns.